File: iosApp/iosApp/Presentation/Cafe/CafeUiState.swift
//
//  CafeUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

struct CafeUiState {
    var isLoading: Bool
    var isLoadingMoreCasts: Bool
    var errorMessage: String?
    var selectedTab: TabType
    var detail: CafeDetail?
    var casts: [CafeDetailCast]
    var castsNextCursor: String?
    var canLoadMoreCasts: Bool
    var isLoadingMenuGoods: Bool
    var hasLoadedMenuGoods: Bool
    var isLoadingMoreNotices: Bool
    var noticesNextCursor: String?
    var canLoadMoreNotices: Bool
    var events: [CafeEventManagementItem]
    var notices: [CafeNoticeManagementItem]
    var isLoadingMoreReviews: Bool
    var reviewsNextCursor: String?
    var canLoadMoreReviews: Bool
    var reviews: [CafeDetailReview]
    var isFavorite: Bool
    var isLoggedIn: Bool
    var isVisitVerified: Bool
    var shouldScrollToTopOnReturn: Bool
    var currentUserId: String?
    var shouldShowFavoriteTooltip: Bool

    static let empty = CafeUiState(
        isLoading: false,
        isLoadingMoreCasts: false,
        errorMessage: nil,
        selectedTab: .info,
        detail: nil,
        casts: [],
        castsNextCursor: nil,
        canLoadMoreCasts: false,
        isLoadingMenuGoods: false,
        hasLoadedMenuGoods: false,
        isLoadingMoreNotices: false,
        noticesNextCursor: nil,
        canLoadMoreNotices: false,
        events: [],
        notices: [],
        isLoadingMoreReviews: false,
        reviewsNextCursor: nil,
        canLoadMoreReviews: false,
        reviews: [],
        isFavorite: false,
        isLoggedIn: false,
        isVisitVerified: false,
        shouldScrollToTopOnReturn: false,
        currentUserId: nil,
        shouldShowFavoriteTooltip: false
    )

    enum TabType: CaseIterable {
        case info
        case casts
        case menu
        case reviews
        case notices
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/cafe/CafeUiState.kt
package com.hhp227.concafe.presentation.cafe

import com.hhp227.concafe.domain.model.CafeDetail
import com.hhp227.concafe.domain.model.CafeDetailCast
import com.hhp227.concafe.domain.model.CafeNoticeManagementItem
import com.hhp227.concafe.domain.model.CafeDetailReview

data class CafeUiState(
    val isLoading: Boolean = false,
    val isLoadingMoreCasts: Boolean = false,
    val errorMessage: String? = null,
    val selectedTab: TabType = TabType.INFO,
    val detail: CafeDetail? = null,
    val casts: List<CafeDetailCast> = emptyList(),
    val castsNextCursor: String? = null,
    val canLoadMoreCasts: Boolean = false,
    val isLoadingMenuGoods: Boolean = false,
    val hasLoadedMenuGoods: Boolean = false,
    val isLoadingMoreNotices: Boolean = false,
    val noticesNextCursor: String? = null,
    val canLoadMoreNotices: Boolean = false,
    val notices: List<CafeNoticeManagementItem> = emptyList(),
    val isLoadingMoreReviews: Boolean = false,
    val reviewsNextCursor: String? = null,
    val canLoadMoreReviews: Boolean = false,
    val reviews: List<CafeDetailReview> = emptyList(),
    val isFavorite: Boolean = false,
    val isLoggedIn: Boolean = false,
    val isVisitVerified: Boolean = false,
    val shouldScrollToTopOnReturn: Boolean = false,
    val currentUserId: String? = null
) {
    enum class TabType {
        INFO,
        CASTS,
        MENU,
        REVIEWS,
        NOTICES
    }

    companion object {
        fun empty(): CafeUiState {
            return CafeUiState()
        }
    }
}
